src/main/java/com/mraof/minestuck/item/crafting/alchemy/generator/SmithingInterpreter.java

```diff
@@ -0,0 +1,75 @@
+package com.mraof.minestuck.item.crafting.alchemy.generator;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import com.mraof.minestuck.item.crafting.alchemy.GristSet;
+import net.minecraft.item.Item;
+import net.minecraft.item.crafting.AbstractCookingRecipe;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.item.crafting.SmithingRecipe;
+import net.minecraft.util.JSONUtils;
+import net.minecraft.util.NonNullList;
+import net.minecraftforge.fml.common.ObfuscationReflectionHelper;
+import net.minecraftforge.registries.ObjectHolder;
+
+import java.lang.reflect.Field;
+
+public class SmithingInterpreter extends DefaultInterpreter
+{
+	public static final SmithingInterpreter INSTANCE = new SmithingInterpreter();
+
+	@ObjectHolder("minestuck:smithing")
+	public static final InterpreterSerializer<SmithingInterpreter> SERIALIZER = null;
+
+	@Override
+	public GristSet generateCost(IRecipe<?> recipe, Item output, GenerationContext context)
+	{
+		// SmithingRecipes don't list their ingredients as ingredients so use this as workaround
+		try
+		{
+			GristSet totalCost = new GristSet();
+
+			Ingredient base = (Ingredient)ObfuscationReflectionHelper.findField(SmithingRecipe.class, "field_234837_a_").get(recipe);
```

Comment:
I'd suggest either caching the field object in a static field, or change to using `ObfuscationReflectionHelper.getPrivateValue()`. The latter is easier to write, but the former is more appropriate under these circumstances.